Data_Engine/Query/AdjacencyDictionary.cs

```diff
@@ -0,0 +1,49 @@
+﻿/*
+ * This file is part of the Buildings and Habitats object Model (BHoM)
+ * Copyright (c) 2015 - 2020, the respective contributors. All rights reserved.
+ *
+ * Each contributor holds copyright over their respective contributions.
+ * The project versioning (Git) records all such contribution source information.
+ *                                           
+ *                                                                              
+ * The BHoM is free software: you can redistribute it and/or modify         
+ * it under the terms of the GNU Lesser General Public License as published by  
+ * the Free Software Foundation, either version 3.0 of the License, or          
+ * (at your option) any later version.                                          
+ *                                                                              
+ * The BHoM is distributed in the hope that it will be useful,              
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of               
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the                 
+ * GNU Lesser General Public License for more details.                          
+ *                                                                            
+ * You should have received a copy of the GNU Lesser General Public License     
+ * along with this code. If not, see <https://www.gnu.org/licenses/lgpl-3.0.html>.      
+ */
+
+using BH.oM.Data.Collections;
+using System.Collections.Generic;
+using System.Linq;
+using System.ComponentModel;
+using BH.oM.Reflection.Attributes;
+
+namespace BH.Engine.Data
+{
+    public static partial class Query
+    {
+        /***************************************************/
+        /**** Public Methods                            ****/
+        /***************************************************/
+        [Description("Gets the adjacency dictionary of a graph, each key value pair in the resulting dictionary is in the form <node, neighbours>")]
+        [Input("graph", "The graph to extract the adjacency dictionary from")]
+        public static Dictionary<GraphNode<T>, List<GraphNode<T>>> AdjacencyDictionary<T>(this Graph<T> graph)
```

Comment:
Would it be more helpful to have an `AdjacencyMatrix` object?
https://en.wikipedia.org/wiki/Adjacency_matrix

Can be an `AdjacencyDictionary` in case, but I think the `AdjacencyMatrix` is one of the standard format to represent graphs, at least in causality, decisions, and particle models.

Reply:
Good point we should include other adjacency representations. Implemented as a dictionary just to support a specific case. Can we put this in the wish list to add later or when the need arises?

Reply:
Yes - I really like the idea of the Adjacency Matrix too going forward. 
@rolyhudson - makes sense to me - I would be happy for this is be raised as issue for future development after this PR.